source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -522,14 +523,46 @@ let parse source_jsons =
     lineage_analysis;
   }
 
-
-let validate { sources; sinks; features; _ } =
+let get_location ?encoding section label (src: [`Channel of in_channel | `String of string]) =
+  let rec loop position section label d = match Jsonm.decode d with
+  | `Lexeme (`Name n) when String.equal n section ->
+    begin match Jsonm.decode d with
+      | `Lexeme (`Name m) when String.equal m label ->
+        begin match Jsonm.decode d with
+          | `Lexeme (`String _) -> loop ((Jsonm.decoded_range d)::position) section label d
+          | _ -> loop position section label d
+        end
+      | _ -> loop position section label d
+    end
+  | `Lexeme _ | `Error _ -> loop position section label d
+  | `End -> List.rev position
+  | `Await -> assert false
+  in
+  loop [] section label (Jsonm.decoder ?encoding src)
+
+let validate { sources; sinks; features; _ } configurations =
   let ensure_list_unique ~kind ~get_name elements =
     let seen = String.Hash_set.create () in
     let ensure_unique element =
       let element = get_name element in
       if Hash_set.mem seen element then
-        failwith (Format.sprintf "Duplicate entry for %s: `%s`" kind element);
+        Log.error "Duplicate entry for %s `%s` found:" kind element in
+        let section = kind ^ "s" in
+        let positions = List.map configurations ~f:(get_location section element) in
+        let rec print_locations locations = match locations with
+        | [] -> ()
+        | location_list::rest -> (
+          match location_list with
+          | [((starting_row, starting_column), (ending_row, ending_column))]
+          | ((starting_row, starting_column), (ending_row, ending_column))::_ -> (
+            Log.error "Entry for `%s` found near lines %d to %d column %d to %d" element starting_row ending_row starting_column ending_column)
+          | [] | _ -> ()
+          in
+          print_locations rest
+        )
+        in
+        print_locations positions;
+        failwith (Format.sprintf "Duplicate entries are not allowed.");
```

Comment:
```suggestion
        Log.error "Duplicate entry for %s `%s` found:" kind element;
        let section = kind ^ "s" in
        let positions = List.map configurations ~f:(get_location section element) in
        let rec print_locations locations =
          match locations with
          | [] -> ()
          | location_list :: rest ->
              (match location_list with
              | [((starting_row, starting_column), (ending_row, ending_column))]
              | ((starting_row, starting_column), (ending_row, ending_column)) :: _ ->
                  Log.error
                    "Entry for `%s` found near lines %d to %d column %d to %d"
                    element
                    starting_row
                    ending_row
                    starting_column
                    ending_column
              | _ -> ());
              print_locations rest
        in
        print_locations positions;
        failwith (Format.sprintf "Duplicate entries are not allowed."));
```

Reply:
Ah @onionymous thank you for taking a look. It makes sense!